docs/examples/addon-restartless/bootstrap.js

```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const {utils: Cu} = Components;
+
+Cu.import('resource://gre/modules/XPCOMUtils.jsm');
+let console = Cu.import('resource://gre/modules/Console.jsm').console;
+
+XPCOMUtils.defineLazyModuleGetter(this, 'Services',
+  'resource://gre/modules/Services.jsm');
+XPCOMUtils.defineLazyModuleGetter(this, 'registerVariants',
+  'chrome://restartless-experiment/content/variants.js');
+
+
+// Define your tests...
+const variantTests = {
+  someFeature: {
+    name: 'Some Feature',
+    description: 'An example A/B test.',
+    variants: [
+      { description: 'A', value: 'A', weight: 1 },
+      { description: 'B', value: 'B', weight: 1 }
+    ]
+  }
+};
+
+function startup(data, reason) {
+  // Then, on startup, ask Test Pilot what to do.
+  registerVariants('restartless-experiment@mozilla.com', variantTests).then(payload => {
+    // When Test Pilot responds, finish initialization with the payload:
+    console.log('Finish initialization with:', JSON.parse(payload.data));
```

Comment:
Should also probably add some code here that fires off a telemetry ping, to exercise the bit that adds variant info to the metrics payload. Took me a few minutes to realize it wasn't actually showing up in the main ping